internal/config/config.go

```diff
@@ -130,3 +104,179 @@ func RootGitRepoDir() (string, error) {
 func ProjectConfigFile() string {
 	return projectConfigName
 }
+
+func readAccessToken() (string, error) {
+	ring, err := openKeyring()
+
+	if errors.Is(err, keyring.ErrNoAvailImpl) {
+		accessToken, tokenErr := readAccessTokenPath()
+		return string(accessToken), tokenErr
+	}
+
+	item, err := ring.Get(keyringKey)
+	if err == nil {
+		return string(item.Data), nil
+	}
+
+	if errors.Is(err, keyring.ErrKeyNotFound) {
+		// Migrate to keychain
+		accessToken, tokenErr := readAccessTokenPath()
+		if len(accessToken) > 0 && tokenErr == nil {
+			return migrateAccessToken(ring, accessToken)
+		}
+		return "", nil
```

Comment:
I feel like this should return an error of some sort, yeah?

Reply:
@mdlayher I thought that originally as well, but that breaks. The reason is that we create a config object always on the CLI startup, also if there's no authentication token configured yet. So right now today an empty token means no token and that case should not error since it breaks things.

It might be good to make that more explicit, but that's I think something for a separate change.

Reply:
Sure, mind just dropping a one-liner TODO comment to note this behavior? An empty, empty return usually looks like a bug.